### PR TITLE
#749  Rename deprecated assert* to require* in o1js precondition APIs 

### PR DESCRIPTION
**Pull request by LuffySama-Dev**
Hi @mitschabaude , @ymekuria , @barriebyron ,

Please find the below changes that I have made to close the issue number #749 .

**Changes Made:**

- Renamed deprecated assert* with require*
- Updated the tutorials package.json to use latest o1js version

**Files/Tutorials changed:**

- Examples from 01 to 11 and Interacting with zkapps are updated
- Tutorials from 01 to Interacting with zkapps are updated
- Other documentations where also updated

**Files/Tutorials Pending to be updated:**

- Facing issue with example 06-offchain-storage. I have updated it locally but the offchain storage zkApp contract's package.json has peer dependency to version of "experimental-offchain-zkapp-storage". So I think we need to update it so, I will need your guidance over here on what to change and how. cc: @mitschabaude 
- Anonymous message board has not used assert for state so no changes made but, I have locally updated it's package.json to use latest o1js version. My question is should I commit it or not ?
- o1js-reference is the only folder I am yet to go through so will make changes to it soon.

Thank You
Luffy 😃

### CODE REVIEW

File: examples/zkapps/01-hello-world/package.json [hunk 1]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/01-hello-world/package.json:46]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/02-private-inputs-and-hash-functions/package.json [hunk 2]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/02-private-inputs-and-hash-functions/package.json:46]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/04-zkapp-browser-ui/contracts/package.json [hunk 3]
```diff
@@ -45,6 +45,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "0.13.*"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/04-zkapp-browser-ui/contracts/package.json:48]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/05-common-types-and-functions/package.json [hunk 4]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/05-common-types-and-functions/package.json:46]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/07-oracles/contracts/package.json [hunk 5]
```diff
@@ -45,6 +45,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "0.13.*"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/07-oracles/contracts/package.json:48]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/08-custom-tokens/package.json [hunk 6]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/08-custom-tokens/package.json:46]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/09-recursion/package.json [hunk 7]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/09-recursion/package.json:46]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/10-account-updates/package.json [hunk 8]
```diff
@@ -44,7 +44,7 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/10-account-updates/package.json:47]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/11-advanced-account-updates/package.json [hunk 9]
```diff
@@ -44,7 +44,7 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/11-advanced-account-updates/package.json:47]:
```suggestion
    "o1js": "0.15.1"
```

File: examples/zkapps/interacting-with-zkApps-server-side/package.json [hunk 10]
```diff
@@ -42,6 +42,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "^0.12.1"
+    "o1js": "0.15.0"
```
Comment (ymekuria) [inline examples/zkapps/interacting-with-zkApps-server-side/package.json:45]:
```suggestion
    "o1js": "0.15.1"
```

**Review by ymekuria — CHANGES_REQUESTED** [review]
Thanks for making these changes @LuffySama-Dev! Could you update the `o1js` version to `0.15.1` in all the tutorials?

**Comment by LuffySama-Dev** [comment]
Hi @ymekuria ,

Yes sure!

Thanks

**Comment by LuffySama-Dev** [comment]
Hi @ymekuria ,

I have made changes as requested.

Thanks 😃

**Review by ymekuria — APPROVED** [review]
Thanks @LuffySama-Dev !

**Pull request merged**